airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2788,6 +2788,8 @@ def execute(self, context: Any):
             impersonation_chain=self.impersonation_chain,
         )
         self.hook = hook
+        if self.project_id is None:
+            self.project_id = hook.project_id
```

Comment:
to be consistent with L2852, although either is fine
```suggestion
        self.project_id = self.project_id or self.hook.project_id
```

Reply:
I would remove all references that do `project_id = self.project_id or self.hook.project_id` since this is what caused the whole issue in the first place, IMO.

The style I'm following is the one used for most hooks assignments when mutating the object on execution